test/unit/range/iterator_test_template.hpp

```diff
@@ -642,6 +651,69 @@ TYPED_TEST_P(iterator_fixture, compare_geq)
     }
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
+// Contiguous Iterator
+// ---------------------------------------------------------------------------------------------------------------------
+
+TYPED_TEST_P(iterator_fixture, address_test)
+{
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    {
+        auto begin_it = std::ranges::begin(this->test_range);
+        EXPECT_EQ(std::to_address(begin_it), std::addressof(*begin_it));
+    }
+
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag> &&
+                  TestFixture::const_iterable)
+    {
+        auto cbegin_it = std::ranges::cbegin(this->test_range);
+        EXPECT_EQ(std::to_address(cbegin_it), std::addressof(*cbegin_it));
+    }
+}
+
+template <typename it_begin_t, typename it_sentinel_t>
+inline void address_difference_test(it_begin_t it_begin, it_sentinel_t it_end)
+{
+    using difference_t = std::iter_difference_t<it_begin_t>;
+
+    for (auto it = it_begin; it != it_end; ++it)
+        EXPECT_EQ(std::to_address(it), std::to_address(it_begin) + static_cast<difference_t>(it - it_begin));
```

Comment:
```suggestion

    for (auto it = it_begin; it != it_end; ++it)
    {
        EXPECT_EQ(std::to_address(it), std::to_address(it_begin) + static_cast<difference_t>(it - it_begin));
        EXPECT_EQ(std::to_address(it), std::to_address(it_end) - static_cast<difference_t>(it_end - it));
    }
```

I think we should also test the address from the sentinel, since it might not be of the same type as the begin iterator.